### PR TITLE
Change Find Command to find by any matching substring

### PR DESCRIPTION
**Pull request by aldentantan**
- As long as the keyword provided to the Find Command matches any substring in the Person's name, that Person will be returned.

### CODE REVIEW

**Comment by aldentantan** [comment]
Closes #58

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 1]
```diff
@@ -31,11 +30,7 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
         checkArgument(!preppedWord.isEmpty(), "Word parameter cannot be empty");
         checkArgument(preppedWord.split("\\s+").length == 1, "Word parameter should be a single word");
 
-        String preppedSentence = sentence;
```
Comment (Johnwz123) [inline src/main/java/seedu/address/commons/util/StringUtil.java:34]: Do you know if there was/what was the original rationale for splitting the sentence to individual words and using streams? If there is some optimisation involved, let's not remove it.

**Review by Johnwz123 — COMMENTED** [review]
See my comment.

File: src/main/java/seedu/address/commons/util/StringUtil.java (hunk 1 above)
Comment (aldentantan) [reply src/main/java/seedu/address/commons/util/StringUtil.java]: Thanks for pointing that out! I believe the rationale was to terminate searching the entire array of words once a matching word was found. I have made the relevant changes to use the Streams implementation instead to keep this behaviour.

**Review by Johnwz123 — APPROVED** [review]
Thanks for making the edit. LGTM!

**Pull request merged**